README.md

```diff
@@ -0,0 +1,219 @@
+      Cookie: cart_id=273
+
+We can look at this HTTP header, get the `cart_id` from it, and look it up using the `ActiveRecord` find method we know and love.
+
+## Security concerns
+
+Cookies are stored as plain text in a user's browser. Therefore, the user can see what's in them, and they can set them to anything they want.
+
+If you open the developer console in your browser, you can see the cookies set by the current site. In Chrome's console, you can find this under `Resources > Cookies`. You can delete any cookie you like. For example, if you delete your `user_session` cookie on `github.com`, you will find that you've been logged out.
+
+You can also edit cookies, for example with [this extension|edit_this_cookie].
+
+This presents a problem for us. If users can edit their `cart_id` cookie, then they can see other users' shopping carts.
+
+## Rails to the rescue
+
+Fortunately, Rails has a solution to this. When you set cookies in Rails, you usually don't manipulate the HTTP headers directly. Instead, you use the `session` method. The `session` method is available anywhere in the Rails response cycle, and it behaves like a hash:
+
+    # set cart_id
+    session[:cart_id] = @cart.id
+
+    # load the cart referenced in the session
+    @cart = session[:cart_id]
+
+You can store any simple Ruby object in the session. In fact, we don't need a `Cart` model at all—we can just store a list of items in the session!
+
+Rails manages all session data in a single cookie, named `_YOUR_RAILS_APP_NAME_session`. It *serializes* your all the key/value pairs you set with `session`, turning them into a big string. Whenever you set a `key` with the `session` method, Rails updates the value of its session cookie to this big string.
+
+When you set cookies this way, Rails signs them to prevent users from tampering with them. Your Rails server has a key, configured in `config/secrets.yml`.
+
+    development:
+      secret_key_base: kaleisgreat  # probably not the most secure key ever
+
+Somewhere else, Rails has a method, let's call it `sign`, which takes a `message` and a `key` and returns a `signature`, which is just a string:
+
+    # sign(message: string, key: string) -> signature: string
+    def sign(message, key):
+      # cryptographic magic here
+      return signature
+    end
+
+It's guaranteed that given the same message and key, sign will produce output. Also, without the key, it is practically impossible to know what `sign` would return for a given message. That is, signatures can't be forged.
+
+Rails creates a signature for every cookie it sets, and appends the signature to the cookie.
+
+When it receives a cookie, Rails verifies that the signature matches the content (that is, that `sign(cookie_body, secret_key_base) == cookie_signature`).
+
+This prevents cookie tampering. If a user tries to edit their cookie and change the `cart_id`, they signature won't match, and Rails will silently ignore the cookie and set a new one.
```

Comment:
Just mention that, if all of the cryptographic magic doesn't make sense, that's ok. What they need to know is that the signature ensures nobody tampered with the cookie.